content/en/docs/concepts/cluster-administration/node-autoscaling.md

```diff
@@ -0,0 +1,276 @@
+manage multiple Node groups. When provisioning, Cluster Autoscaler will add Nodes to the group that
+best fits the requests of pending Pods. When consolidating, Cluster Autoscaler always selects
+specific Nodes to remove as opposed to just resizing the underlying VM group.
+
+Additional context:
+
+* Documentation overview:
+  https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/README.md
+* Supported cloud providers:
+  https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/README.md#faqdocumentation
+* Extensive FAQ: https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md
+* Contact: https://github.com/kubernetes/community/tree/master/sig-autoscaling#contact
+
+#### Karpenter
+
+Karpenter auto-provisions VMs based on [NodePool](https://karpenter.sh/docs/concepts/nodepools/)
+configuration provided by the user. Karpenter handles all aspects of node lifecycle, not just
+autoscaling. It works directly with VMs and doesn't rely on provider-side VM groups.
+
+Additional context:
+
+* Documentation: https://karpenter.sh/
+* Supported cloud providers:
+  https://github.com/kubernetes-sigs/karpenter?tab=readme-ov-file#supported-cloudproviders
+* Extensive FAQ: https://karpenter.sh/docs/faq/
+* Contact:
+  https://github.com/kubernetes-sigs/karpenter#community-discussion-contribution-and-support
+
+#### Implementation comparison
+
+Main differences between Cluster Autoscaler and Karpenter:
+
+* Cluster Autoscaler provides features related to just Node autoscaling. Karpenter has a wider
+  scope, and also provides features intended for managing Node lifecycle altogether (for example
+  grouping Nodes into pools, handling Node upgrades via disruption).
+* Cluster Autoscaler doesn't support auto-provisioning, the Node groups it can provision from have
+  to be pre-configured. Karpenter supports auto-provisioning, so the user only has to configure a
+  set of constraints for the provisioned Nodes instead of fully configuring homogenous groups.
+* Cluster Autoscaler provides integrations with numerous cloud providers, including smaller and less
+  popular providers. Karpenter currently only provides integrations with AWS and Azure.
+
+## Configure workloads for autoscaling {#workload-config}
+
+Normally, workloads don't have to be additionally configured to work with Node autoscaling, beyond
+just configuring the autoscaler. However, autoscalers provide the ability to change their behavior
+based on workload characteristics. The configuration options documented here are supported by all
+official autoscalers. Thanks to this, the workloads can be portable between clusters using different
+autoscalers. Additionally, the autoscalers provide their own, non-portable configuration options
+(see autoscaler-specific documentation).
+
+### Configure a workload not to be disrupted by consolidation
+
+{{< feature-state state="stable" for_k8s_version="v1.31" >}}
```

Comment:
```suggestion
```
This feature was not part of the Kubernetes v1.31 release, because that release hasn't happened, and also because (AIUI) there's no in-tree code that honors this annotation.

Reply:
You need to either remove this line, or delay this PR until v1.31 has been released and then rebase.

Reply:
Understood, I'll remove the section before submitting and follow-up in a separate PR, just wanted some initial feedback from the Karpenter folks on this.